search/includes/classes/queue/class-cron.php

```diff
@@ -64,7 +64,8 @@ public function init() {
 		// Add the custom cron schedule
 		add_filter( 'cron_schedules', [ $this, 'filter_cron_schedules' ], 10, 1 );
 
-		$this->schedule_sweeper_job();
+		// Hook into init action to ensure cron-control has already been loaded
+		add_action( 'init', [ $this, 'schedule_sweeper_job' ] );
```

Comment:
Can we do this on admin_init? Scheduling cron on the front-end triggers a cron SELECT and potentially an UPDATE on every request.

Reply:
Actually, this pattern looks pretty good:

https://github.com/Automattic/Cron-Control/blob/fd02cfbef233698c93ab9afb444d25b54a7a3fef/includes/class-internal-events.php#L42-L48